runtime/src/main/java/org/corfudb/runtime/view/replication/QuorumReplicationProtocol.java

```diff
@@ -86,7 +86,7 @@ public ILogData peek(Layout layout, long address) {
                 return null;
             }
             if (readResponse != null) {
-                LogData result = readResponse.getReadSet().get(address);
+                LogData result = readResponse.getAddresses().get(address);
```

Comment:
If we do not have , then we need test when the readList is less than batch size, at batch size, greater than batch size, mulitple batches. Also if not there, we might need to add negative tests, when one or more batches fail (transient) .